creator-node/src/snapbackSM/secondarySyncHealthTracker.js

```diff
@@ -41,6 +41,34 @@ const SecondarySyncHealthTracker = {
     }
   },
 
+  async computeUserSecondarySyncSuccessRates (wallet, secondary1, secondary2) {
+    // Retrieve map of all SyncRequestOutcome keys and daily counts for user from all secondaries
+    const userSecondarySyncHealthOutcomes = await SecondarySyncHealthTracker.getSyncMetricsForUser(wallet)
+
+    // Compute sync success rate per secondary for user
+    let sec1UserSyncSuccesses = 0; let sec1UserSyncFailures = 0; let sec2UserSyncSuccesses = 0; let sec2UserSyncFailures = 0
+    for (let [key, count] of Object.entries(userSecondarySyncHealthOutcomes)) {
```

Comment:
does this create an iterable each time? can we instantiate the iterable before this for loop and use it here

Reply:
not sure i follow but if you mean that `Object.entries(userSecondarySyncHealthOutcomes)` is re-computed each loop iteration - i don't know for sure but i'd be very very surprised if that was the case, wouldn't really make sense

Reply:
just checked and it doesn't. we good here